src/lib/zf/private/zf_emu_superbuf.h

```diff
@@ -16,6 +16,8 @@
 /* prevent ci/driver/ci_efct.h and ci_aux.h from getting included */
 #define CI_DRIVER_CI_EFCT_H
 
+#include <zf_internal/private/zf_emu.h>
```

Comment:
I'm not 100% sure what you mean, but since Xilinx-CNS/onload_internal@575a8f572eb20af4af68765ff1ddd8856f38567c we are using the `__DECLARE_FLEX_ARRAY` macro in userspace which is fine since it is part of the uapi. However, the kernel compat is [`#ifndef` guarded against c++](https://github.com/Xilinx-CNS/onload_internal/blob/onload-8.1/src/include/ci/tools/sysdep.h#L66), so if we try to build tcpdirect on a RHEL7 machine it won't have a definition.

In commit 7589b95 I added a redefinition of `__DECLARE_FLEX_ARRAY` to `zf_emu.c` to get tcpdirect building again since gcc doesn't like having a flexible array member in the middle of a struct, more details in #17

Reply:
Thanks for referring to https://github.com/Xilinx-CNS/tcpdirect/commit/7589b95054779c3610490e92e5714085dc5a426e in https://github.com/Xilinx-CNS/tcpdirect/tree/v8_1. I was looking at this file in the wrong branch (master) and didn't realise it is defined over there.